tools/fixedpointarithmetic/src/main/java/dk/alexandra/fresco/fixedpoint/DefaultFixedNumeric.java

```diff
@@ -0,0 +1,147 @@
+package dk.alexandra.fresco.fixedpoint;
+
+import dk.alexandra.fresco.framework.DRes;
+import dk.alexandra.fresco.framework.builder.numeric.ProtocolBuilderNumeric;
+import dk.alexandra.fresco.framework.value.SInt;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.RoundingMode;
+
+public class DefaultFixedNumeric implements FixedNumeric {
+
+  private final int precision;
+  private final ProtocolBuilderNumeric builder;
+
+  /**
+   * Creates a FixedNumeric which creates basic numeric operations
+   *
+   * @param builder The protocolbuilder used to construct underlying protocols.
+   * @param precision  Amount of digits after the dot.
+   */
+  public DefaultFixedNumeric(ProtocolBuilderNumeric builder, int precision) {
+    super();
+    this.precision = precision;
+    this.builder = builder;
+  }
+  
+  @Override
+  public DRes<SFixed> known(BigDecimal value) {
```

Comment:
Wondering if double suffices?

int is capped after 32 bits but double can represent a lot more with the 64 bits.

Reply:
It probably is, however then we need to manually handle scales and double <-> BigInteger conversion.

Reply:
Makes sense - let us keep it like this for now